dashboard/lib/dynamic_config/gatekeeper.rb

```diff
@@ -0,0 +1,81 @@
+# A module that allows us to dynamically turn on and off
+# features without requiring a code push.
+
+require 'dynamic_config/datastore_cache'
+require 'dynamic_config/adapters/dynamodb_adapter'
+require 'dynamic_config/adapters/json_file_adapter'
+require 'dynamic_config/adapters/memory_adapter'
+
+class GatekeeperBase
+  def initialize(datastore_cache)
+    @datastore_cache = datastore_cache
+  end
+
+  # @param feature [String] the name of the feature
+  # @param where [Hash] a hash of conditions
```

Comment:
Could you provide some examples in the documentation of how the "conditions" work?  This is a pretty powerful concept so not self-evident.